.github/workflows/test.yml

```diff
@@ -58,3 +58,7 @@ jobs:
     - name: run regression tests
       run: |
        make test
+    - name: code coverage
+      run: |
+       PYTHON="coverage run" make test_unittest
+       make coverage_report
```

Comment:
So, we run unit tests first, then all tests (including unit tests) — because 'coverage_report' target depends of 'test' target. But the statistics will be collected only from the first run of the unit tests.

Reply:
Personally, I like the way when we gather code coverage for unit testing and system tests separately. But if we want to use Coveralls we need to combine coverage data for a common report so I updated patch for workflow and Makefile:
```
--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,4 @@
+MAKEFILE_PATH := $(abspath $(lastword $(MAKEFILE_LIST)))
 TEST_RUN_EXTRA_PARAMS?=
 PYTHON?=python
 
@@ -13,11 +14,15 @@ luacheck:
        luacheck --config .luacheckrc .
 
 test_integration:
-       $(PYTHON) test/test-run.py --force $(TEST_RUN_EXTRA_PARAMS)
+       PYTHON="$(PYTHON)" test/test-run.py --force $(TEST_RUN_EXTRA_PARAMS)
 
 test_unittest:
        $(PYTHON) -m unittest discover test/unittest/
 
 test: test_unittest test_integration
 
+coverage:
+       make -f $(MAKEFILE_PATH) test
+       coverage report --skip-empty --precision=1
+
 .PHONY: lint flake8 luacheck test test_integration test_unittest
```

```
--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -58,3 +58,8 @@ jobs:
     - name: run regression tests
       run: |
        make test
+    - name: code coverage
+      run: |
+       make coverage
+       pip install coveralls
+       coveralls
```

Reply:
Okay, thanks!